web/controllers/user_controller.ex

```diff
@@ -11,6 +11,22 @@ defmodule CoursePlanner.UserController do
     render(conn, "index.html", users: Users.all())
   end
 
+  def show(%{assigns: %{current_user: %{role: role}}} = conn, %{"id" => id}) do
+    if role === "Coordinator" do
```

Comment:
This is not necessary as we are already restricting access to this controller in `permissions.ex`.
If you tested with other roles and you could access it, that's a bug and we should fix there, not here (it shouldn't even call this function).